docs/supported-patterns.md

```diff
@@ -0,0 +1,82 @@
+    result2 = yield ctx.call_activity(hello, input='Seattle')
+    result3 = yield ctx.call_activity(hello, input='London')
+
+    return [result1, result2, result3]
+```
+
+You can find the full sample [here](./examples/activity_sequence.py).
+
+### Fan-out/fan-in
+
+An orchestration can fan-out a dynamic number of function calls in parallel and then fan-in the results using the following syntax:
+
+```python
+# activity function for getting the list of work items
+def get_work_items(ctx: task.ActivityContext, _) -> List[str]:
+    # ...
+
+# activity function for processing a single work item
+def process_work_item(ctx: task.ActivityContext, item: str) -> int:
+    # ...
+
+# orchestrator function that fans-out the work items and then fans-in the results
+def orchestrator(ctx: task.OrchestrationContext, _):
+    # the number of work-items is unknown in advance
+    work_items = yield ctx.call_activity(get_work_items)
+
+    # fan-out: schedule the work items in parallel and wait for all of them to complete
+    tasks = [ctx.call_activity(process_work_item, input=item) for item in work_items]
+    results = yield task.when_all(tasks)
+
+    # fan-in: summarize and return the results
+    return {'work_items': work_items, 'results': results, 'total': sum(results)}
+```
+
+You can find the full sample [here](./examples/fanout_fanin.py).
+
+### Human interaction and durable timers
+
+An orchestration can wait for a user-defined event, such as a human approval event, before proceding to the next step. In addition, the orchestration can create a timer with an arbitrary duration that triggers some alternate action if the external event hasn't been received:
+
+```python
+def purchase_order_workflow(ctx: task.OrchestrationContext, order: Order):
+    """Orchestrator function that represents a purchase order workflow"""
+    # Orders under $1000 are auto-approved
+    if order.Cost < 1000:
+        return "Auto-approved"
+
+    # Orders of $1000 or more require manager approval
+    yield ctx.call_activity(send_approval_request, input=order)
+
+    # Approvals must be received within 24 hours or they will be canceled.
+    approval_event = ctx.wait_for_external_event("approval_received")
+    timeout_event = ctx.create_timer(timedelta(hours=24))
+    winner = yield task.when_any([approval_event, timeout_event])
+    if winner == timeout_event:
+        return "Canceled"
+
+    # The order was approved
+    yield ctx.call_activity(place_order, input=order)
+    approval_details = approval_event.get_result()
+    return f"Approved by '{approval_details.approver}'"
+```
+
+As an aside, you'll also notice that the example orchestration above works with custom business objects. Support for custom business objects includes support for custom classes, custom data classes, and named tuples. Serialization and deserialization of these objects is handled automatically by the SDK.
+
+You can find the full sample [here](./examples/human_interaction.py).
```

Comment:
Do all the links to samples need to be fixed now that you've changed the location of this content?
```suggestion
You can find the full sample [here](../examples/human_interaction.py).
```